File: 07.function/4673_self_number.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    int n=10000;
    int d;
    string s;
    vector<bool> check(n+1, true);

    for (int a=1; a<=n; ++a) {
        d = a;
        for (auto c : to_string(a))
            d += int(c)-48;
        check[d] = false;

        if (check[a])
            cout << a << endl;
    }

    return 0;
}

File: 07.function/4673_self_number.py
#!/usr/bin/env python

n = 10000
check = [True for _ in range(n+1)]

for a in range(1,n+1):
    d = a + sum([int(c) for c in str(a)])
    if d <= n:
        check[d] = False

    if check[a]:
        print(a)
